ts/Series/GeoHeatmap/GeoHeatmapSeries.ts

```diff
@@ -29,12 +40,146 @@ const {
 } = SeriesRegistry;
 
 import U from '../../Core/Utilities.js';
+import PointerEvent from '../../Core/PointerEvent.js';
+import Point from '../../Core/Series/Point.js';
+import type MapView from '../../Maps/MapView.js';
 
 const {
+    addEvent,
     extend,
-    merge
+    isNumber,
+    merge,
+    pick
 } = U;
 
+import IU from '../InterpolationUtilities.js';
+
+const {
+    colorFromPoint,
+    getContext
+} = IU;
+
+/**
+ * Normalize longitute value to -180:180 range.
+ * @private
+ */
+function normalizeLonValue(lon: number): number {
+    return lon - Math.floor((lon + 180) / 360) * 360;
+}
+
+/**
+ * Get proper point's position for PixelData array.
+ * @private
+ */
+function scaledPointPos(
+    lon: number,
+    lat: number,
+    canvasWidth: number,
+    canvasHeight: number,
+    colsize: number,
+    rowsize: number
+): number {
+    return Math.ceil(
+        (canvasWidth * (canvasHeight - 1 - (lat + 90) / rowsize)) +
+        ((lon + 180) / colsize)
+    );
+}
+
+/**
+ * Project ImageData to actual mapView projection used on a chart.
+ * @private
+ */
+function getProjectedImageData(
```

Comment:
Functions defined like this (scope) are not wrappable or overwritable from outside making any hotfix or any potential extension / wrap longer. getProjectedImageData could be a method of GeoHeatmapSeries instead.